GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -0,0 +1,425 @@
+            this.mount = false;
+            this.ExitCode = ReturnCode.Success;
+        }
+
+        public UpgradeOrchestrator()
+        {
+            string logFilePath = GVFSEnlistment.GetNewGVFSLogFileName(
+                ProductUpgrader.GetLogDirectoryPath(),
+                GVFSConstants.LogFileTypes.UpgradeProcess);
+            JsonTracer jsonTracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "UpgradeProcess");
+            jsonTracer.AddLogFileEventListener(
+                logFilePath,
+                DefaultEventLevel,
+                Keywords.Any);
+
+            this.tracer = jsonTracer;
+            this.preRunChecker = new InstallerPreRunChecker(this.tracer, GVFSConstants.UpgradeVerbMessages.GVFSUpgradeConfirm);
+            this.upgrader = new ProductUpgrader(ProcessHelper.GetCurrentProcessVersion(), this.tracer);
+            this.output = Console.Out;
+            this.input = Console.In;
+            this.mount = false;
+            this.ExitCode = ReturnCode.Success;
+        }
+
+        public ReturnCode ExitCode { get; private set; }
+
+        public void Execute()
+        {
+            string error = null;
+
+            ProductUpgrader.RingType ring = ProductUpgrader.RingType.Invalid;
+
+            if (!this.TryLoadUpgradeRing(out ring, out error) || ring == ProductUpgrader.RingType.None)
+            {
+                string message = ring == ProductUpgrader.RingType.None ?
+                    GVFSConstants.UpgradeVerbMessages.NoneRingConsoleAlert :
+                    GVFSConstants.UpgradeVerbMessages.InvalidRingConsoleAlert;
+                this.output.WriteLine(message);
+
+                if (ring == ProductUpgrader.RingType.None)
+                {
+                    this.output.WriteLine(GVFSConstants.UpgradeVerbMessages.SetUpgradeRingCommand);
+                }
+            }
+            else
+            {
+                try
+                {
+                    Version newVersion = null;
+                    if (!this.TryRunUpgrade(out newVersion, out error))
+                    {
+                        this.ExitCode = ReturnCode.GenericError;
+                    }
+                }
+                finally
+                {
+                    string mountError = null;
+                    if (!this.TryMountRepositories(out mountError))
+                    {
+                        mountError = Environment.NewLine + "WARNING: " + mountError;
+                        this.output.WriteLine(mountError);
+                    }
+
+                    this.DeletedDownloadedAssets();
+                }
+            }
+
+            if (this.ExitCode == ReturnCode.GenericError)
+            {
+                error = Environment.NewLine + "ERROR: " + error;
+                this.output.WriteLine(error);
+            }
+            else
+            {
+                this.output.WriteLine(Environment.NewLine + "Upgrade completed successfully!");

```

Comment:
I still find this output confusing, and I think users will be similarly wondering if this is an error or a success:

```
Mounting repositories...Failed

WARNING: GVFS service --mount-all failed.

Upgrade completed successfully!
Press Enter to exit.
```

I think that final success message needs to acknowledge that mount failed, but also that upgrading the product succeeded. Otherwise people will just think this is a bug and will be emailing us about it.

Reply:
<img width="507" alt="screen shot 2018-10-03 at 3 42 06 pm" src="https://user-images.githubusercontent.com/378580/46443199-76cd5980-c73a-11e8-9246-9682e0b20529.png">

Reply:
Updated the messaging.

<img width="766" alt="screen shot 2018-10-04 at 11 43 38 am" src="https://user-images.githubusercontent.com/378580/46543640-5a8f0100-c88f-11e8-892e-5b50024f256a.png">